raw_exec windows: add support for setting the task user (#25496)
hashicorp/nomad: website/content/docs/drivers/raw_exec.mdx

  conflict with other Nomad driver's cgroups and have unintended side
  effects.

> ~> On Linux the `task.user` field cannot be set on a Task using the `raw_exec` driver if
> the Nomad client has been hardened according to the [production][hardening] guide.
> On Windows, when Nomad is running as [system service][service] less privileged service
> user can be specified (example: `NT AUTHORITY\LocalService`, `NT AUTHORITY\NetworkService`)

Reviewer: ```suggestion
~> On Linux, you cannot set the `task.user` field on a task using the `raw_exec`
driver if you have hardened the Nomad client according to the
[production][hardening] guide. On Windows, when Nomad is running as a [system
service][service], you may specify a less-privileged service user. For example,
`NT AUTHORITY\LocalService`, `NT AUTHORITY\NetworkService`.
```

Style guide: changed passive voice to active
